core/modules/block/block.module

```diff
@@ -418,8 +418,9 @@ function block_custom_block_save(array $edit, $delta = NULL, $langcode = NULL) {
   // See https://github.com/backdrop/backdrop-issues/issues/2137.
   $fids = filter_parse_file_fids($edit['body']['value']);
   $files = file_load_multiple($fids);
+  $status_permanent = FILE_STATUS_PERMANENT;
   foreach ($files as $fid => $file) {
-    if ($file && $file->status !== FILE_STATUS_PERMANENT) {
+    if ($file && $file->status !== "$status_permanent") {
```

Comment:
Took me bit to understand what you did. I was thinking that something like this might be the more the typical approach and takes up fewer lines:

```suggestion
    if ((int) ($file && $file->status) !== FILE_STATUS_PERMANENT) {
```

Reply:
Yep, I thought about that way, too, but then decided against it, because that could be ambiguous. Constant FILE_STATUS_PERMANENT is the integer `1`. Think of something like `(int) TRUE` ... yikes. String to string comparison is most reliable here.

Reply:
Funny, I was thinking `(int) TRUE` and `(int) FALSE` would be pretty unambiguous. They can only be 1 or 0.

Whichever, way we go I think it should reflect the typical way this is treated in Backdrop. Perhaps we can find an another example?

Reply:
I found one example. Ironically, I think it was added by you @indigoxela :smile: 

```
function form_process_html_datetime($element) {
  $value = $element['#value'];
  // Conversion to a timestamp has happened, possibly in a preview.
  // Get it back into an array for the form.
  if (is_numeric($value) && (int) $value == $value) {
```

Reply:
Here's another I think.

```
function image_rotate_dimensions(array &$dimensions, array $data) {
  // If the rotate is not random and the angle is a multiple of 90 degrees,
  // then the new dimensions can be determined.
  if (!$data['random'] && ((int) ($data['degrees']) == $data['degrees']) && ($data['degrees'] % 90 == 0)) {
```

Reply:
I guess they're not exactly the same. But they do cast booleans to integers.

Reply:
You convinced me. :wink: